### ISSUE TITLE
Waterstone Design Brainstorming

### ISSUE THREAD

**Issue by CovertJaguar** [Railcraft/Railcraft#850]
So the Waterstone is similar to the Firestone in that it is a large rechargeable Fluid Container filled with Water. It was intended to be implemented alongside the Firestone but I could never come up with a suitable recharge ritual that wasn't too cheaty for the finite water players and didn't leave behind water craters. Recent discussions have revisited the idea, revolving around the concept of a boss fight to recharge the stone.

**General concepts:**
- Ore Found in Water Temples
- Refined with Diamond Picks, Prismarine, Sponge?
- Unrefined stones spawn non-source water randomly when held
- Fluid Container, can be drained only
- Massive water capacity, like ocean in your pocket (need suggestions for numbers)
- Spawn source water block on right click (like a bucket)
- Right click on lava creates obsidian
- Grants Aqua Affinity, Respiration, and Depth Strider to holder when held in off-hand/main-hand

**Recharge Ritual:**
1. Toss it in an ocean
2. Raises above surface on column of water
3. Sucks up water from ocean creating crater in surface
4. Sucks player up to orbit around stone, grants Respiration
5. Large sphere of water contained by ice forms around stone/player
6. Elder Guardian spawns, when he dies he drops the recharged Waterstone
7. Sphere collapses back into the ocean, ocean surface is repaired to level

**Alternate Recharge Ritual:**
1. Toss stone onto sheet of ice
2. Stone absorbs ice blocks and charges 1 point per block (Compressed ice could be worth 5-10 points per block)
This alternate method is slower and requires lots of ice to charge the stone fully, but doesn't require fighting.

**Other Ideas**
- Holding both an unrefined waterstone and unrefined firestone spawn a cloud of steam around player, player immune?
- Sword crafting for special weapons
- New sounds for both stones!

**Comment by Forecaster** [comment]
I'd go with 5000 capacity, same as the firestone for easy remembering mainly.

**Comment by CovertJaguar** [comment]
Is that in buckets? Because 5k mB is way too small.

**Comment by bookerthegeek** [comment]
Buckets I would think.

**Comment by Forecaster** [comment]
Would be in buckets, yeah, same as 1 lava block = 1 point for the firestone

**Comment by Forecaster** [comment]
Also, alternate charging ritual:
1. Toss stone onto sheet of ice
2. Stone absorbs ice blocks and charges 1 point per block (Compressed ice could be worth 5-10 points per block)
This alternate method is slower and requires lots of ice to charge the stone fully, but doesn't require fighting.

**Comment by bookerthegeek** [comment]
Has the added benefit or working in peaceful mode.

**Comment by CovertJaguar** [comment]
Good point!

**Comment by bookerthegeek** [comment]
Just to make sure, this would be able to fill any tank with a gui correct? If so, that is amazing bro-tato!

**Comment by CovertJaguar** [comment]
That's the point.

**Comment by Forecaster** [comment]
More brainstorming from IRC:
- Swords crafted from firestone & waterstone respectively, grants same abilities as the stone, plus weapons are enchanted and has right-click ability (eg firebolts/water lance or something), but cannot be used as fuel/drained or used to affect in-world blocks anymore.

**Comment by Peach774** [comment]
What exactly is the point though? Water in MC is already infinite. You already have the water tower to generate infinite water. Shouldnt is be more useful somehow? Maybe icestone vs waterstone, i dunno, just brainstorming.

**Comment by Vexatos** [comment]
@Peach774 Water isn't infinite if you don't want it to be.

**Comment by Peach774** [comment]
@Vexatos but should a new thing be added that will only be used by a few players of the mod?

**Comment by Forecaster** [comment]
That's a stupid argument. Have you tried fitting a water tower in a locomotive? : P

**Comment by Peach774** [comment]
Tank cart?

**Comment by Forecaster** [comment]
Why do you care if an item is added? Other uses have been suggested besides just being a water source. And even if it wasn't it'd vastly surpass the capacity of a tank cart.

**Comment by bookerthegeek** [comment]
Or boilers!

**Comment by purpleposeidon** [comment]
The stones could do with some sound effects. Here's some stuff from a couple minutes of browsing freesounds:
Fire: http://freesound.org/people/midimagician/sounds/249418/
Water: http://freesound.org/people/blaukreuz/sounds/201930/
They're both licensed under CC0, which is equivalent to public domain.

**Comment by purpleposeidon** [comment]
Rain is a wet thing that hasn't been mentioned yet.

**Comment by Peach774** [comment]
That might have some use. Being able to cause rain with one and stop rain with the other.

**Comment by KnightMiner** [comment]
I agree causing rain with the waterstone would be a good idea, though stopping it with the firestone sounds a bit odd, as fire removing rainclouds sounds odd.

Would it be odd if that was the ability for holding an unprocessed waterstone instead of random puddles?

**Comment by purpleposeidon** [comment]
The most interesting part of the firestone is the journey to obtaining it, quite an epic. Scouring the nether ocean, bringing it back to base while it sets everything on fire, spending 12 diamonds to cut it, and then returning to the nether to charge it. Water may be a lesser element in Minecraft, but why not give the Waterstone a similarly epic story?

The ice spikes biomes might've been mentioned on IRC. (How much of a chore is finding that biome? Eye of Frost; it's an Eye of Ender that leads to the nearest ice spikes biome? Pearl of Frost, it straight-up teleports you?) So, perhaps some of the ice spikes contain a core of Frozen-Solid Waterstone Ore, which must be defrosted at the cost of a raw firestone. (Possible hint text: "Nothing short of a raging inferno could melt this.") So this piggybacks on some of the firestone story. So next you might give it a pass through the rock crusher. And later you take it to the ocean boss fight/ice farm.

Perhaps frozen-solid waterstone ore has a super-cool texture, and it's not so rare that you couldn't use them as accent blocks or something, but the firestone requirement makes it not an easy thing to mass produce. But if it's common then maybe there shouldn't be an easy way of finding ice spikes biomes.

**Comment by CovertJaguar** [comment]
That sounds like a cool back story for the Icestone.

**Comment by liach** [comment]
I like icestone actually.

**Comment by liach** [comment]
Now vanilla adds conduit. We do not prefer duplicate features.

**Comment by Generalcamo** [comment]
Irrelevant: the conduit is an underwater beacon, not a way to store huge amounts of water.

**Comment by CovertJaguar** [comment]
Actually... I think the conduit could be a useful thing to tap into for help with charging a Waterstone.

Such as it needs to be used on a conduit built at a certain depth beneath the surface of an ocean biome. Doing so spawns some mobs to fight.

**Comment by liach** [comment]
Can we require the shells in 1.13 to recharge the waterstone?

**Comment by CovertJaguar** [comment]
Oh....the Heart of the Sea would be a great thing from which to craft the Waterstone.

I'm not sure shells are common enough.

**Comment by liach** [comment]
shells are rare drops of fishing and the drowned mob. it's easy to find hearts if you explore enough of the map, as there are many shipwrecks.
